pkg/controller/datavolume-controller.go

```diff
@@ -616,7 +637,7 @@ func (r *DatavolumeReconciler) reconcileDataVolumeStatus(dataVolume *cdiv1.DataV
 
 func (r *DatavolumeReconciler) emitEvent(dataVolume *cdiv1.DataVolume, dataVolumeCopy *cdiv1.DataVolume, curPhase cdiv1.DataVolumePhase, event *DataVolumeEvent) error {
 	// Only update the object if something actually changed in the status.
-	if !reflect.DeepEqual(dataVolume.Status, dataVolumeCopy.Status) {
+	if !reflect.DeepEqual(dataVolume, dataVolumeCopy) {
```

Comment:
Won't this cause unwanted updates now?

Reply:
The controller shouldn't be updating the spec anywhere.  And if it is, it would be great to know and stop it from doing that.